transaction.proto

```diff
@@ -15,12 +16,14 @@ message Fee {
     JubJubCompressed R = 3;
     JubJubCompressed pk_r = 4;
 }
+
 message Note {
-    JubJubCompressed randomness = 1;
-    JubJubCompressed pk_r = 2;
-    JubJubCompressed commitment = 3;
-    BlsScalar nonce = 4;
-    PoseidonCipher encypted_data = 5;
+    int32 note_type = 1;
```

Comment:
Not sure about that. We unified them in phoenix_core, so why do we specify ghe type here? Can you check it with @ZER0 ?

Reply:
The struct in `phoenix-core` still has this identifier too though. I reckon we should keep it, otherwise we would need to infer based on the underlying data whether or not a note is obfuscated/transparent, and that sounds like a needless hassle. I'm gonna guess this is the same reason that the identifier still exists in `phoenix-core`, but I'll let @ZER0 weigh in on that.

Reply:
@CPerezz exactly because they're unified we need to specify the type there, otherwise you will have two different messages, `TransparentNote` and `ObfuscatedNote`. Since, instead, we have one single `Note` we need a field to indicate the type.

However I would prefer using an `enum` in the schema:
```
message Note {
    enum NoteType {
        TRANSPARENT = 0;
        OBFUSCATED = 1;
    }
    NoteType note_type = 1;
   // etc
}